modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -18,10 +18,10 @@ export class {{classname}} {
     protected basePath = '{{basePath}}';
     public defaultHeaders : Headers = new Headers();
 
-    constructor(protected http: Http, basePath: string) {
```

Comment:
@wing328 I fixed constructor and added  `@Optional` decorator to bashPath argument.